phenol-io/src/main/java/org/monarchinitiative/phenol/io/annotations/hpo/OrphanetInheritanceXMLParser.java

```diff
@@ -0,0 +1,230 @@
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+import static org.monarchinitiative.phenol.formats.hpo.HpoModeOfInheritanceTermIds.*;
+import static org.monarchinitiative.phenol.formats.hpo.HpoModeOfInheritanceTermIds.INHERITANCE_ROOT;
+import static org.monarchinitiative.phenol.formats.hpo.HpoModeOfInheritanceTermIds.OLIGOGENIC;
+
+public class OrphanetInheritanceXMLParser {
+  /**
+   * Path to {@code en_product9_age.xml} file.
+   */
+  private final String orphanetXmlPath;
+  /**
+   * Reference to the HPO Ontology.
+   */
+  private final Ontology ontology;
+  /**
+   * A String of the form ORPHA:orphadata[2019-01-05] that we will use as the biocuration entry.
+   */
+  private final String orphanetBiocurationString;
+  /**
+   * Key: an Orphanet disease id; value: an array list of HpoAnnotations, one for each inheritance mode
+   * that is associated with the disease.
+   */
+  private final Multimap<TermId, HpoAnnotationEntry> disease2inheritanceMultimap;
+  // XML Parsing
+  private static final String DISORDER = "Disorder";
+  private static final String ORPHA_NUMBER = "OrphaNumber";
+  private static final String TYPE_OF_INHERITANCE_LIST = "TypeOfInheritanceList";
+  private static final String TYPE_OF_INHERITANCE = "TypeOfInheritance";
+  private static final String NAME = "Name";
+  private static final String AVERAGE_AGE_OF_ONSET_LIST = "AverageAgeOfOnsetList";
+  private static final String AVERAGE_AGE_OF_DEATH_LIST = "AverageAgeOfDeathList";
+  private static final String DISORDER_TYPE = "DisorderType";
+  /**
+   * Orphanet marks some of its intheritance entries as Not apploicable. We will just skip them.
+   * This is the corresponding ID.
+   */
+  private static final String NOT_APPLICABLE_ID = "409941";
+  /**
+   * similar to above. We will skip this.
+   */
+  private static final String UNKNOWN_ID = "409939";
+  /**
+   * similar to above. We will skip this.
+   */
+  private static final String NO_DATA_AVAILABLE = "409940";
+
+
+  private boolean inDisorder = false;
+  private boolean inTypeOfInheritanceList = false;
+  private boolean inTypeOfInheritance = false;
+  private boolean inAverageAgeOfOnsetList = false;
+  private boolean isInAverageAgeOfDeathList = false;
+  private boolean inDisorderType = false;
+
+  public OrphanetInheritanceXMLParser(String xmlpath, Ontology onto) {
+    orphanetXmlPath = xmlpath;
+    this.ontology = onto;
+    String todaysDate = getTodaysDate();
+    orphanetBiocurationString = String.format("ORPHA:orphadata[%s]", todaysDate);
+    disease2inheritanceMultimap = ArrayListMultimap.create();
+    parse(new File(orphanetXmlPath));
+  }
+
+
+  public Multimap<TermId, HpoAnnotationEntry> getDisease2inheritanceMultimap() {
+    return disease2inheritanceMultimap;
+  }
+
+  private void parse(File file) {
+    try {
```

Comment:
Doesn't this need to be wrapped using a try-with-resources? Otherwise you've not closed the file having read it.

Reply:
fixed